telegram/auth.go

```diff
@@ -0,0 +1,59 @@
+package telegram
+
+import (
+	"bufio"
+	"context"
+	"errors"
+	"fmt"
+	"github.com/gotd/td/telegram/auth"
+	"github.com/gotd/td/tg"
+	"golang.org/x/crypto/ssh/terminal"
+	"os"
+	"strings"
+)
+
+// noSignUp can be embedded to prevent signing up.
+type noSignUp struct{}
+
+func (c noSignUp) SignUp(ctx context.Context) (auth.UserInfo, error) {
+	return auth.UserInfo{}, errors.New("not implemented")
+}
+
+func (c noSignUp) AcceptTermsOfService(ctx context.Context, tos tg.HelpTermsOfService) error {
+	return &auth.SignUpRequired{TermsOfService: tos}
+}
+
+// termAuth implements authentication via terminal.
+type TermAuth struct {
+	noSignUp
+
+	phone string
+}
+
+func (a TermAuth) Phone(_ context.Context) (string, error) {
```

Comment:
We don't need this argument, why do we add it? same below.

Reply:
This function user tg's login support